### PR TITLE
Support GRE L2/L3 tunneling protocol.

### PR DESCRIPTION
**Pull request by tinoue**
This PR supports GRE tunneling protocol over tailscale network.

### CODE REVIEW

File: net/packet/gre4.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
```
Comment (Xe) [inline net/packet/gre4.go:1]:
```suggestion
// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
```

File: net/packet/gre6.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
```
Comment (Xe) [inline net/packet/gre6.go:1]:
```suggestion
// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
```

**Comment by DentonGentry** [comment]
Could you add Signoff lines to the commit descriptions?
https://github.com/tailscale/tailscale/pull/2162/checks?check_run_id=2880344338

File: wgengine/filter/tailcfg.go [hunk 3]
```diff
@@ -18,6 +18,7 @@ var defaultProtos = []ipproto.Proto{
 	ipproto.UDP,
 	ipproto.ICMPv4,
 	ipproto.ICMPv6,
+	ipproto.GRE,
```
Comment (bradfitz) [inline wgengine/filter/tailcfg.go:21]: we don't want to add GRE to the defaultProtos. It needs to be opt-in for users, like we did for SCTP.

Comment (tinoue) [reply wgengine/filter/tailcfg.go:21]: @bradfitz 
Thank you for your comment.
To make GRE as opt-in option is nice idea.

I checked about SCTP and it seems to be allowed by port based ACL.
The problem is GRE doesn't have any port - like ICMP - and current ACL doesn't have protocol based rule.
Do you have any idea ?

**Comment by amosbird** [comment]
Is there any plan to merge this PR?

**Comment by tinoue** [comment]
@amosbird Thanks for reminding me about this PR. Closed becuase the latest tailscale already supports any type of protocol.

**Comment by amosbird** [comment]
@tinoue Great. Starting from which version does tailscale have this ability?

**Comment by tinoue** [comment]
@amosbird  From 1.2.0 according to https://github.com/tailscale/tailscale/issues/2163#issuecomment-997146054
You can specify any protocol byTailscale ACL rule (see proto field): https://tailscale.com/kb/1018/acls/#acl-rules

**Pull request closed without merging**